packages/components/ng-ovh-otrs/src/otrs-popup/otrs-popup.controller.js

```diff
@@ -0,0 +1,543 @@
+        OTRS_POPUP_INTERVENTION_ENUM.REPLACEMENTDISK &&
+      self.formDetails === 'message'
+    ) {
+      self.refreshRequests();
+      self.setForm('start');
+    }
+  };
+
+  function getServerInfo() {
+    self.loaders.intervention = true;
+    self.intervention.canHotSwap = false;
+    self.intervention.hasMegaRaid = false;
+    self.intervention.slotInfo.canUseSlotId = null;
+    self.intervention.slotInfo.slotsCount = 0;
+    return OtrsPopupInterventionService.getServerInterventionInfo(
+      self.intervention.serviceName.serviceName,
+    )
+      .then((serverInfo) => {
+        self.intervention.canHotSwap = serverInfo.canHotSwap;
+        self.intervention.hasMegaRaid = serverInfo.hasMegaRaid;
+        self.intervention.slotInfo = serverInfo.slotInfo;
+        if (self.intervention.slotInfo.canUseSlotId) {
+          self.intervention.enums.slotID = times(
+            self.intervention.slotInfo.slotsCount,
+          );
+        }
+      })
+      .catch(() => {
+        manageAlert(
+          $translate.instant('otrs_intervention_disk_error'),
+          'danger',
+        );
+      })
+      .finally(() => {
+        self.loaders.intervention = false;
+      });
+  }
+
+  self.addDisk = () => {
+    const newItemNo = self.intervention.disk.disks.length + 1;
+    self.intervention.disk.disks.push({ id: newItemNo });
+  };
+
+  self.removeChoice = (item) => {
+    self.intervention.disk.disks.splice(item, 1);
+  };
+
+  self.setForm = (formDetails) => {
+    self.formDetails = formDetails;
+  };
+
+  self.isSalesCategory = () =>
+    self.ticket.category === OTRS_POPUP_CATEGORIES.SALES;
+
+  self.continueForm = () => {
+    if (
+      self.ticket &&
+      self.ticket.category === OTRS_POPUP_CATEGORIES.INTERVENTION &&
+      self.ticket.subcategory === OTRS_POPUP_INTERVENTION_ENUM.REPLACEMENTDISK
+    ) {
+      self.intervention.serviceName = self.ticket.serviceName;
+      self.setForm(OTRS_POPUP_INTERVENTION_ENUM.REPLACEMENTDISK);
+    } else if (
+      self.ticket &&
+      self.ticket.category === OTRS_POPUP_CATEGORIES.INTERVENTION &&
+      self.ticket.subcategory === OTRS_POPUP_INTERVENTION_ENUM.OTHER
+    ) {
+      self.ticket.category = OTRS_POPUP_CATEGORIES.INCIDENT;
+      self.ticket.subcategory = OTRS_POPUP_INCIDENT_ENUM.DOWN;
+      self.setForm('message');
+    } else {
+      self.setForm('message');
+    }
+  };
+
+  this.$onInit = () => {
+    initFields();
+
+    self.alert = {
+      visible: false,
+      type: null,
+      message: null,
+    };
+
+    self.servicesValues = OTRS_POPUP_SERVICES;
+    self.formDetails = 'start';
+    self.interventionEnum = OTRS_POPUP_INTERVENTION_ENUM;
+
+    self.baseUrlTickets = OtrsPopup.getBaseUrlTickets();
+
+    if (isEmpty(self.baseUrlTickets)) {
+      throw new Error('A baseUrlTickets must be specified.');
+    }
+
+    return $q
+      .all({
+        translations: $injector.invoke(/* @ngTranslationsInject:json ./translations */),
```

Comment:
Since translations files has moved from XML to JSON, I've added the format to the
following annotation `@ngTranslationsInject:json`.